### PR TITLE
modernize per analyzers

### PR DESCRIPTION
**Pull request by DickBaker**
the existing .SLN raises Warnings/Messages and analyzers encourage modernizing changes.

This PR offers cleaner code, so I hope MS owners will accept these changes as-is
but **Allow edits by maintainers** checkbox set to allow tweaking

### CODE REVIEW

**Comment by DickBaker** [comment]
sorry, my first commit only touched 1 file and rest stayed behind. Hopefully this 2nd attempt will work for all changed files.

**Comment by guardrex** [comment]
Thanks @DickBaker ... You dropped the `var` keyword in favor of explicit types. We follow (most of) the ASP.NET Core framework coding guidelines, and the use of `var` with implicit types is a coding guideline that we follow here, even if the analyzer isn't happy about it 😄. Those lines must be reverted. After that, I'll get Jeremy in to take a look, as this is his sample app. Everything else looks *GrEaT* to me! 👍

BTW --- If reverting the `var` lines is too much of a chore, I'll take care of it.

File: 8.0/BlazorWebAppEFCore/DatabaseUtility.cs [hunk 1]
```diff
@@ -10,10 +11,10 @@ public static async Task EnsureDbCreatedAndSeedWithCountOfAsync(DbContextOptions
     {
         // Empty to avoid logging while inserting (otherwise will flood console).
         var factory = new LoggerFactory();
-        var builder = new DbContextOptionsBuilder<ContactContext>(options)
+        DbContextOptionsBuilder<ContactContext> builder = new DbContextOptionsBuilder<ContactContext>(options)
```
Comment (guardrex) [inline 8.0/BlazorWebAppEFCore/DatabaseUtility.cs:14]:
```suggestion
        var builder = new DbContextOptionsBuilder<ContactContext>(options)
```

File: 8.0/BlazorWebAppEFCore/DatabaseUtility.cs [hunk 2]
```diff
@@ -10,10 +11,10 @@ public static async Task EnsureDbCreatedAndSeedWithCountOfAsync(DbContextOptions
...
             .UseLoggerFactory(factory);
 
-        using var context = new ContactContext(builder.Options);
+        await using ContactContext context = new(builder.Options);
```
Comment (guardrex) [inline 8.0/BlazorWebAppEFCore/DatabaseUtility.cs:17]:
```suggestion
        using var context = new ContactContext(builder.Options);
```

File: 8.0/BlazorWebAppEFCore/Grid/GridQueryAdapter.cs [hunk 3]
```diff
@@ -52,29 +51,25 @@ public async Task<ICollection<Contact>> FetchAsync(IQueryable<Contact> query)
     {
         query = FilterAndQuery(query);
         await CountAsync(query);
-        var collection = await FetchPageQuery(query)
+        List<Contact> collection = await FetchPageQuery(query)
```
Comment (guardrex) [inline 8.0/BlazorWebAppEFCore/Grid/GridQueryAdapter.cs:54]:
```suggestion
        var collection = await FetchPageQuery(query)
```

File: 8.0/BlazorWebAppEFCore/Grid/GridQueryAdapter.cs [hunk 4]
```diff
@@ -86,20 +81,20 @@ private IQueryable<Contact> FilterAndQuery(IQueryable<Contact> root)
         // Apply a filter?
         if (!string.IsNullOrWhiteSpace(_controls.FilterText))
         {
-            var filter = _filterQueries[_controls.FilterColumn];
+            Func<IQueryable<Contact>, IQueryable<Contact>> filter = _filterQueries[_controls.FilterColumn];
```
Comment (guardrex) [inline 8.0/BlazorWebAppEFCore/Grid/GridQueryAdapter.cs:84]:
```suggestion
            var filter = _filterQueries[_controls.FilterColumn];
```

File: 8.0/BlazorWebAppEFCore/Grid/GridQueryAdapter.cs [hunk 5]
```diff
@@ -86,20 +81,20 @@ private IQueryable<Contact> FilterAndQuery(IQueryable<Contact> root)
...
             sb.Append($"Filter: '{_controls.FilterColumn}' ");
             root = filter(root);
         }
 
         // Apply the expression.
-        var expression = _expressions[_controls.SortColumn];
+        Expression<Func<Contact, string>> expression = _expressions[_controls.SortColumn];
```
Comment (guardrex) [inline 8.0/BlazorWebAppEFCore/Grid/GridQueryAdapter.cs:90]:
```suggestion
        var expression = _expressions[_controls.SortColumn];
```

File: 8.0/BlazorWebAppEFCore/Program.cs [hunk 6]
```diff
@@ -1,9 +1,9 @@
 using BlazorWebAppEFCore.Components;
-using Microsoft.EntityFrameworkCore;
 using BlazorWebAppEFCore.Data;
 using BlazorWebAppEFCore.Grid;
+using Microsoft.EntityFrameworkCore;
 
-var builder = WebApplication.CreateBuilder(args);
+WebApplicationBuilder builder = WebApplication.CreateBuilder(args);
```
Comment (guardrex) [inline 8.0/BlazorWebAppEFCore/Program.cs:6]:
```suggestion
var builder = WebApplication.CreateBuilder(args);
```

File: 8.0/BlazorWebAppEFCore/Program.cs [hunk 7]
```diff
@@ -26,14 +26,14 @@
 // Service to communicate success on edit between pages
 builder.Services.AddScoped<EditSuccess>();
 
-var app = builder.Build();
+WebApplication app = builder.Build();
```
Comment (guardrex) [inline 8.0/BlazorWebAppEFCore/Program.cs:29]:
```suggestion
var app = builder.Build();
```

File: 8.0/BlazorWebAppEFCore/Program.cs [hunk 8]
```diff
@@ -26,14 +26,14 @@
...
 
 // This section sets up and seeds the database. Seeding is NOT normally
 // handled this way in production. The following approach is used in this
 // sample app to make the sample simpler. The app can be cloned. The
 // connection string is configured. The app can be run.
-await using var scope = app.Services.GetRequiredService<IServiceScopeFactory>().CreateAsyncScope();
-var options = scope.ServiceProvider.GetRequiredService<DbContextOptions<ContactContext>>();
+await using AsyncServiceScope scope = app.Services.GetRequiredService<IServiceScopeFactory>().CreateAsyncScope();
```
Comment (guardrex) [inline 8.0/BlazorWebAppEFCore/Program.cs:35]:
```suggestion
await using var scope = app.Services.GetRequiredService<IServiceScopeFactory>().CreateAsyncScope();
```

File: 8.0/BlazorWebAppEFCore/Program.cs [hunk 9]
```diff
@@ -26,14 +26,14 @@
...
+DbContextOptions<ContactContext> options = scope.ServiceProvider.GetRequiredService<DbContextOptions<ContactContext>>();
```
Comment (guardrex) [inline 8.0/BlazorWebAppEFCore/Program.cs:36]:
```suggestion
var options = scope.ServiceProvider.GetRequiredService<DbContextOptions<ContactContext>>();
```

**Review by guardrex — CHANGES_REQUESTED** [review]

**Comment by DickBaker** [comment]
**Luke**, thanks for doing the "var" dance to save me the effort. As part of my refactoring I had applied the "_Analyze and Code Cleanup_", "_Run Code Cleanup (profile 1)_" with my settings to apply ALL the .editorconfig particulars. My VS would have input the .editorconfig from the repo's primitive
https://github.com/dotnet/blazor-samples/blob/main/.editorconfig 17 lines
versus
https://github.com/dotnet/runtime/blob/main/.editorconfig 190 lines
https://github.com/dotnet/docs/blob/main/.editorconfig  210 lines
https://github.com/dotnet/aspnetcore 475 lines

Obviously the .editorconfig is there to herd devs into some standard pens,
so my [**strong**!] suggestion is that you cherry-pick the bits _you_ like, e.g.
csharp_style_var_for_built_in_types = true
csharp_style_var_when_type_is_apparent = true
csharp_style_var_elsewhere = false

but I don't dare to propose any more subjective electives: over to Jeremy/your team to argue the specifics!

**Comment by guardrex** [comment]
`.editorconfig` was placed by Robert Haken with the basic settings. I've been busy, and it's a low priority. I opened #345 to work on it later, hopefully by EOY. Thanks for your remarks because it probably would've sat for much longer had you not said something.

**Comment by DickBaker** [comment]
yup, I see MVP Robert Haken's commit back in Feb, and I reckon he was equally hesitant to tell you owners how to suck eggs (but nice to have European inputs), so best of luck for you/Jeremy rolling the dice for gory details (I will watch #341 from the touchlines). Maybe loop-in Robert anyway as prev contributor?

**Comment by guardrex** [comment]
Jeremy looked it over and said (via email) that it looks good.

*Merging now!* 🎉 Thanks again!

**Pull request merged**